test/framework/k8s_utils.go

```diff
@@ -371,6 +373,24 @@ func DownloadKubeconfig(ctx context.Context, client kubernetes.Interface, namesp
 	return nil
 }
 
+// DownloadAdminKubeconfigForShoot requests an admin kubeconfig for the given shoot and writes the kubeconfig to the
+// given download path. The kubeconfig expires in 24 hours.
+func DownloadAdminKubeconfigForShoot(ctx context.Context, client kubernetes.Interface, shoot *gardencorev1beta1.Shoot, downloadPath string) error {
+	const expirationSeconds = 24 * 3600 // 24h
+	kubeconfig, err := access.RequestAdminKubeconfigForShoot(ctx, client, shoot, pointer.Int64(expirationSeconds))
+	if err != nil {
+		return err
+	}
+
+	if downloadPath != "" {
+		err = os.WriteFile(downloadPath, []byte(kubeconfig), 0755)
```

Comment:
kubeconfig returned by `RequestAdminKubeconfigForShoot` is already a byte slice.
```suggestion
		err = os.WriteFile(downloadPath, kubeconfig, 0755)
```